src/resmom/requests.c

```diff
@@ -5005,3 +5059,52 @@ req_del_hookfile(struct batch_request *preq) /* ptr to the decoded request   */
 
 	reply_ack(preq);
 }
+
+#if defined(PBS_SECURITY) && (PBS_SECURITY == KRB5)
+void
+req_cred(struct batch_request *preq) /* ptr to the decoded request   */
+{
+	unsigned char	out_data[CRED_DATA_SIZE];
+	ssize_t		out_len = 0;
+	char		buf[LOG_BUF_SIZE];
+	krb5_data	*data;
+	char		*data_base64 = NULL;
+	job		*pjob;
+
+	if (decode_block_base64((unsigned char *)preq->rq_ind.rq_cred.rq_data, preq->rq_ind.rq_cred.rq_size, out_data, &out_len, buf, LOG_BUF_SIZE) != 0) {
+		log_err(errno, __func__, buf);
+		req_reject(PBSE_SYSTEM, 0, preq);
+		return;
+	}
+
+	if ((data = (krb5_data *)malloc(sizeof(krb5_data))) == NULL) {
+		log_err(errno, __func__, "Unable to allocate Memory!\n");
+		req_reject(PBSE_SYSTEM, 0, preq);
+		return;
+	}
+
+	if ((data->data = (char *)malloc(sizeof(unsigned char)*out_len)) == NULL) {
+		log_err(errno, __func__, "Unable to allocate Memory!\n");
+		req_reject(PBSE_SYSTEM, 0, preq);
+		return;
+	}
+
+	data->length = out_len;
+	memcpy(data->data, out_data, out_len);
+
+	data_base64 = strdup(preq->rq_ind.rq_cred.rq_data);
+
+	store_or_update_cred(preq->rq_ind.rq_cred.rq_jobid, preq->rq_ind.rq_cred.rq_credid, preq->rq_ind.rq_cred.rq_type, data, data_base64, preq->rq_ind.rq_cred.rq_validity);
+
+	/* renew ticket for the job*/
+	if ((pjob = find_job(preq->rq_ind.rq_cred.rq_jobid)) != NULL) {
+		/* send cred to sisters too */
+		send_cred_sisters(pjob);
+
+		/* new creds received - lets renew cred */
+		renew_job_cred(pjob);
+	}
+
+	reply_ack(preq);
```

Comment:
i think we need to free data and data->data before returning.

Reply:
Same as above. store_or_update_cred() does not copy data and data->data but it stores it as it. and delete_cred() will free both later.